### DISCUSSION TITLE
Transferring Data to iOS App

### DISCUSSION THREAD

**Discussion by MechtEngineer** [iSoron/uhabits#1359] [Help & FAQ]
Hi! Does anyone know of an iOS app that would be able to import the CSV data exported from Loop Habit Tracker? I'm moving to iOS, and I would like to build on approximately 2 years of data in my habits.

**Comment by iSoron** [comment]
App recommendations are welcome, as long as they are open source. Recommendations to closed source apps will be removed.

**Comment by tomer-ha** [comment]
I am facing a similar problem. Unfortunately I couldn't find any open source alternatives to Loop on iOS.
I hope to be able to invest some time in creating an alternative (or work towards porting Loop), but in the meantime I needed a quick solution to not lose my data. I found a closed-source alternative and created an open source data converter, available [here](https://github.com/tomer-ha/LoopToIosConverter).
Once Loop is available on iOS / I create another app / I return to Android, I plan to re-use the tool to convert the data back from iOS to the new tool.

**Reply by GuyGoldenberg** [reply]
Amazing work!! Thank for for investing your time into this and sharing the solution with everyone.
 Just migrated all of my data to the other app. Unfortunately it's far from being as good as Loop but that's what we have for now until Loop will be available to iOS.

**Reply by raffomania** [reply]
Very nice work, thank you so much. I was really missing loop on my iphone.
